Optional dependencies not met

<!-- *Before creating an issue please make sure you are using the latest version of yarn.* -->

**Do you want to request a *feature* or report a *bug*?**
A bug

**What is the current behavior?**
When running `yarn` optional dependencies of packages are not met. 

**If the current behavior is a bug, please provide the steps to reproduce.**
Run a package with `optionalDependencies` declared
**What is the expected behavior?**
Since there  is an explicit `--ignore-optional` flag, I expect optional dependencies to be met. 

**Please mention your node.js, yarn and operating system version.**
node: 4.1.2
yarn: 0.15.1
osx: 10.12

I'm hitting the same issue via `gulp-imagemin`. It throws this error when I try to run my gulp project that uses that package:

```sh
Error: Cannot find module 'imagemin-gifsicle'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/path/to/project/node_modules/gulp-imagemin/index.js:8:26)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/path/to/project/gulp/images.js:5:18)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
```

I'm not sure if this is a `gulp-imagemin` issue or `imagemin` itself.

same here for `gulp-imagemin`
```
module.js:341                                                                                                 
    throw err;                                                                                                
    ^                                                                                                         
                                                                                                              
Error: Cannot find module 'imagemin-gifsicle'                                                                 
    at Function.Module._resolveFilename (module.js:339:15)                                                    
    at Function.Module._load (module.js:290:25)                                                               
    at Module.require (module.js:367:17)                                                                      
    at require (internal/module.js:16:19)                                                                     
    at Object.<anonymous> (D:\Projects\test\angular2test\node_modules\gulp-imagemin\index.js:8:26)            
    at Module._compile (module.js:413:34)                                                                     
    at Object.Module._extensions..js (module.js:422:10)                                                       
    at Module.load (module.js:357:32)                                                                         
    at Function.Module._load (module.js:314:12)                                                               
    at Module.require (module.js:367:17)                                                                      
```

please note even running `yarn add gulp-imagemin` explicitly doenst work. There is a warning about `fsevents` failed, which is an optional dependency 
```
PS D:\Projects\test\angular2test> yarn add gulp-imagemin
yarn add v0.15.1
[1/4] Resolving packages...
[2/4] Fetching packages...
warning fsevents@1.0.14: The platform "win32" is incompatible with this module.
info "fsevents@1.0.14" is an optional dependency and failed compatibility check. Excluding it from installation.
warning benchmark@1.0.0: The engine "rhino" appears to be invalid.
warning lodash@2.4.2: The engine "rhino" appears to be invalid.
warning lodash@1.0.2: The engine "rhino" appears to be invalid.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
success Saved 1 new dependency
└─ gulp-imagemin@3.0.3
Done in 400.80s.
PS D:\Projects\test\angular2test> gulp install
module.js:341
    throw err;
    ^

Error: Cannot find module 'imagemin-gifsicle'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (D:\Projects\test\angular2test\node_modules\gulp-imagemin\index.js:8:26)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
```

Installing the optional dependency directly `yarn add imagemin-gifsicle` produces below error
```
yarn add v0.15.1
warning angular-2-test@0.0.0: "dependencies" has dependency "gulp-imagemin" with range "^3.0.3" that collides with a dependency in "devDependencies" of the same name with version "3.0.3"
[1/4] Resolving packages...
[2/4] Fetching packages...
warning fsevents@1.0.14: The platform "win32" is incompatible with this module.
info "fsevents@1.0.14" is an optional dependency and failed compatibility check. Excluding it from installation.
warning benchmark@1.0.0: The engine "rhino" appears to be invalid.
warning lodash@2.4.2: The engine "rhino" appears to be invalid.
warning lodash@1.0.2: The engine "rhino" appears to be invalid.
[3/4] Linking dependencies...
error ENOENT: no such file or directory, open 'D:\Projects\test\angular2test\node_modules\@types\angular\index.d.ts'
    at Error (native)
info Visit http://yarnpkg.com/en/docs/cli/add for documentation about this command.
```

But when i run the command `yarn add imagemin-gifsicle` again it installed successfully, so this might be a whole new issue as I encountered the same for `imagemin-jpegtran` and `imagemin-optipng` which are all optional dependency of `gulp-imagemin` I will open a new ticket for this

--- a/src/registries/yarn-registry.js
+++ b/src/registries/yarn-registry.js
@@ -24,7 +24,7 @@ export const DEFAULTS = {
 
   'save-prefix': '^',
   'ignore-scripts': false,
-  'ignore-optional': true,
+  'ignore-optional': false,
   registry: YARN_REGISTRY,
   'user-agent': [
     `yarn/${pkg.version}`,

